eq_translations/survey_schema.py

```diff
@@ -1,245 +1,204 @@
-                question = self.get_parent_question(pointer)
+        for pointer_dict in self.translatable_strings:
+            message_context = pointer_dict.get("context")
+            value = pointer_dict["value"]
 
-                message_context = "Answer for: {}".format(
-                    question["text"] if is_placeholder(question) else question
-                )
+            if not value:
+                continue
 
-                catalog.add(
-                    dumb_to_smart_quotes(pointer_contents),
-                    context=message_context,
-                    auto_comments=[f"answer-id: {parent_answer_id}"],
-                )
+            message_id = get_message_id(content=value)
+            catalog.add(
+                id=message_id, context=message_context,
+            )
 
-        print(f"Total Messages: {total_translations}")
+        print(f"Total Messages: {self.total_translatable_strings}")
 
         return catalog
 
-    def translate(self, schema_translation):
+    def translate(self, schema_translation, language_code):
         """
-        Use the supplied schema translation object to translate pointers found
+        Use the supplied schema translation object and language code to translate pointers found
         within the survey
 
-        :param schema_translation:
+        :param schema_translation: The SchemaTranslation object
+        :param language_code: The target language code
         :return:
         """
         translated_schema = copy.deepcopy(self.schema)
-        total_translations = len(self.no_context_pointers + self.context_pointers)
         missing_translations = 0
 
-        for pointer in self.no_context_pointers:
-            pointer_contents = resolve_pointer(self.schema, pointer)
-            if pointer_contents:
-                translation = schema_translation.translate_message(pointer_contents)
-                if translation:
-                    translated_schema = set_pointer(
-                        translated_schema, pointer, translation
-                    )
-                else:
-                    missing_translations += 1
-                    print(
-                        "Missing translation at {}: '{}'".format(
-                            pointer, pointer_contents
+        for pointer_dict in self.translatable_strings:
+            pointer = pointer_dict["pointer"]
+            value = pointer_dict["value"]
+
+            if not value:
+                continue
+
+            message_id = get_message_id(value)
+            message_context = pointer_dict.get("context")
+
+            translation = schema_translation.get_translation(
+                message_id, message_context
+            )
+
+            if translation:
+                if isinstance(translation, tuple) and language_code:
+                    plural_forms = get_plural_forms_for_language(language_code)
+                    for idx, plural in enumerate(plural_forms):
```

Comment:
Is idx a random identifier or does it represent something?

Reply:
It's to signify 'index', I seen `idx` used commonly instead of `index` but I have changed it to `index` for clarity.